### ISSUE TITLE
Remove `calculateInitialCatalogHookForm()` from create/edit connection form 

### ISSUE THREAD

**Issue by dizel852** [airbytehq/airbyte#31389]
### Relevant information
Relates to https://github.com/airbytehq/airbyte-platform-internal/pull/8835

From the [comment](https://github.com/airbytehq/airbyte-platform-internal/pull/8835#discussion_r1355097934):

> I'm almost sure that we can remove this entire calculateInitialCatalogHookForm as the backend should be doing this for us. (That should be heavily tested, and I don't feel strongly one way or the other about doing it in this PR vs. another)
Commented this file out + manually tested locally with a Postgres source and saw all correct sync modes, cursors, and primary keys applied without us doing any of this!

**Comment by dizel852** [comment]
@HeetVekariya since this one relates to unpublished changes we will solve it internally, thanks!

**Comment by tealjulia** [comment]
Refining notes:
- blocked by the react-hook-form migration of SyncCatalog linked above

**Comment by dizel852** [comment]
Ideally, this task should be completed after both of the following tasks:

[Issue 8639](https://github.com/airbytehq/airbyte-platform-internal/issues/8639)
[Issue 32207](https://github.com/airbytehq/airbyte/issues/32207)

**Comment by dizel852** [comment]
@josephkmh @timroes @tealjulia 

💡 Investigated further:

It seems we still need to perform some initial calculations to remove old data (cursor or PR):

https://github.com/airbytehq/airbyte-platform-internal/blob/22c80d3e48511aa02a37e3b6a778dc33441146fb/oss/airbyte-webapp/src/components/connection/ConnectionForm/hookFormConfig.tsx#L124-L140

After "Refresh schema" and obtaining a catalog diff with breaking changes, we can still observe values that should not be present in the catalog and the backend response.

<img width="1759" alt="Screenshot at Dec 05 14-31-13" src="https://github.com/airbytehq/airbyte/assets/20929344/594d1223-e363-4d3b-9f3d-f96576ce280d">

Comparison with old initial calculations:

<img width="910" alt="image" src="https://github.com/airbytehq/airbyte/assets/20929344/b8091077-1bae-43c3-ab5b-fae1ff2c715d">

Possible solutions:

- Request the BE team to tweak the response and fully rely on that data.
- Continue with some initial calculations, although it contradicts the issue's intent.

Any thoughts? 🤔

**Comment by josephkmh** [comment]
It feels a bit strange to rely on `calculateInitialCatalogHookForm()` to apply updates from a schema refresh. It seems like this function is being used for two different actions: calculating initial form values, and updating the form values based on a schema refresh.

Would it be feasible to pull the schema refresh _out_ of this function (and out of the `ConnectionFormService`/`ConnectionEditService`? IMO this should be a distinct action:
1. Call the BE API to refresh the schema
2. Show the user the diff in the modal
3. Update the form values programmatically (_not_ reinitializing the form initial values)

But maybe I'm missing some reason why these all needs to be tied together in a single context.

**Comment by dizel852** [comment]
> It feels a bit strange to rely on calculateInitialCatalogHookForm() to apply updates from a schema refresh. It seems like this function is being used for two different actions: calculating initial form values and updating the form values based on a schema refresh.

`calculateInitialCatalogHookForm()` was used to apply the optimal sync mode, set initial data (such as the cursor field), and track `breaking` changes. If a field used as a primary key or cursor was removed, we needed to eliminate it from the stream's `config` and render the entire form as invalid. I provided an example where our schema had `breaking` changes, and "refresh schema" was used just to fetch an updated schema. 😅

I've submitted a [PR](https://github.com/airbytehq/airbyte-platform-internal/pull/10129) where I extracted all implementations related to `breaking` changes.
